### PR TITLE
fixing summary

### PR DESCRIPTION
**Pull request by anurag6**

### CODE REVIEW

File: forch/faucet_state_collector.py [hunk 1]
```diff
@@ -97,6 +108,22 @@ def get_dataplane_state(self):
         self._fill_egress_state(dplane_state)
         return dplane_state
 
+    def _get_broken_sw_dp(self, dplane_state):
```
Comment (grafnu) [inline forch/faucet_state_collector.py]: ..._sw_dps (plural)

File: forch/faucet_state_collector.py [hunk 2]
```diff
@@ -97,6 +108,22 @@ def get_dataplane_state(self):
...
+        broken_sw = []
+        sw_map = dplane_state.get(TOPOLOGY_DP_MAP, {})
+        for switch, state in sw_map.items():
+            if state.get(SW_STATE) == constants.STATE_DOWN:
+                broken_sw.append(switch)
+        return broken_sw
+
+    def _get_broken_link_dp(self, dplane_state):
```
Comment (grafnu) [inline forch/faucet_state_collector.py]: ..._link_dps (plural)

File: forch/faucet_state_collector.py [hunk 3]
```diff
@@ -82,9 +82,20 @@ def __init__(self):
     def get_dataplane_summary(self):
         """Get summary of dataplane"""
         dplane_state = self.get_dataplane_state()
+        egress_state = dplane_state.get(EGRESS_STATE)
+        state = constants.STATE_HEALTHY
+        detail = "egress state: "+egress_state
```
Comment (grafnu) [inline forch/faucet_state_collector.py]: construct as list of strings, and then do a "; " join at the end.

Comment (grafnu) [inline forch/faucet_state_collector.py]: Spaces around + -- lint should be complaining!

File: forch/faucet_state_collector.py [hunk 4]
```diff
@@ -97,6 +108,22 @@ def get_dataplane_state(self):
...
+            if state.get(SW_STATE) == constants.STATE_DOWN:
+                broken_sw.append(switch)
+        return broken_sw
+
+    def _get_broken_link_dp(self, dplane_state):
+        broken_links = []
+        link_map = dplane_state.get(TOPOLOGY_LINK_MAP, {})
+        for link, link_obj in link_map.items():
+            if link_obj.get(LINK_STATE) == constants.STATE_DOWN:
+                broken_links.append(link)
```
Comment (grafnu) [inline forch/faucet_state_collector.py]: How "ugly" is this output?  IRRC each link is some ungodly long string that is difficult to read... ?  I'm worried that if there's more than one of them, it'll be near impossible to decyher.

Maybe consider doing a .replace(site_name_prefix, '') on the link name?  That was, nz-kiwi-t1sw2:9 (or whatever it is), gets replaced with the shorter (but still unique) t1sw2:9

or at least something to consider. thoughts?

File: forch/faucet_state_collector.py [hunk 5]
```diff
@@ -82,9 +82,20 @@ def __init__(self):
     def get_dataplane_summary(self):
         """Get summary of dataplane"""
         dplane_state = self.get_dataplane_state()
+        egress_state = dplane_state.get(EGRESS_STATE)
```
Comment (grafnu) [inline forch/faucet_state_collector.py]: Is all this same information represented in the dataplane_state API somewhere?  I know the egress_state is, but what about the others?  I think you should likely extract all this logic (or bits of the logic) to a separate function that can be used as well with the main API... We want/need the from from summary to actual API to be coherent... ?

File: forch/faucet_state_collector.py (hunk 1 above)
Comment (anurag6) [reply forch/faucet_state_collector.py]: its more like "get_broken_switches_for_dp_state" but abbreviated.

**Pull request merged**